3-reveal/migrations/6-MDA/Zambia-2020/deploy/mda_structures.psql

```diff
@@ -0,0 +1,101 @@
+-- Deploy reveal_mda_zambia_2020:mda_structures to pg
+-- requires: reveal_transaction_tables:plans
+-- requires: reveal_transaction_tables:events
+-- requires: reveal_transaction_tables:jurisdictions
+-- requires: reveal_transaction_tables:clients
+-- requires: reveal_transaction_tables:locations
+-- requires: reveal_transaction_tables:tasks
+-- requires: reveal_database_views:jurisdictions_materialized_view
+
+BEGIN;
+
+SET search_path TO :"schema",public;
+
+CREATE OR REPLACE VIEW ntd_dispense_structures AS
+SELECT DISTINCT ON (locations.id, plans.identifier)
+    locations.id AS location_id,
+    plans.identifier AS plan_id,
+    jurisdictions.jurisdiction_id AS jurisdiction_id,
+    jurisdictions.jurisdiction_parent_id AS jurisdiction_parent_id,
+    jurisdictions.jurisdiction_name AS jurisdiction_name,
+    jurisdictions.jurisdiction_depth AS jurisdiction_depth,
+    jurisdictions.jurisdiction_path AS jurisdiction_path,
+    jurisdictions.jurisdiction_name_path AS jurisdiction_name_path,
+    -- COALESCE(mda_dispence_query.nPzqDistribute, 0) AS nPzqDistribute,
+    -- COALESCE(mda_dispence_query.availableForTreatment, 0) AS availableForTreatment,
+    -- COALESCE(mda_dispence_query.nPzqDistributedQuantity, 0) AS nPzqDistributedQuantity,
+    COALESCE(client_info_query.nsac, 0) AS nsac,
+    COALESCE(client_info_query.visited, 0) AS structure_visited
+FROM locations AS locations
+LEFT JOIN plan_jurisdiction AS plan_jurisdiction
+    ON locations.jurisdiction_id = plan_jurisdiction.jurisdiction_id
+LEFT JOIN plans AS plans
+    ON plan_jurisdiction.plan_id = plans.identifier
+    AND plans.intervention_type = 'Dynamic-MDA'
+    AND plans.status IN ('active', 'complete')
+LEFT JOIN jurisdictions_materialized_view AS jurisdictions
+    ON locations.jurisdiction_id = jurisdictions.jurisdiction_id
+LEFT JOIN LATERAL (
+    SELECT
+        COUNT(DISTINCT t1.identifier) AS visited,
+        COALESCE(SUM(COALESCE(c1.attributes->>'nsac', '0')::INTEGER), 0) AS nsac
+    FROM tasks AS t1
+    LEFT JOIN events AS e2
+        ON t1.identifier = e2.task_id
+    LEFT JOIN clients AS c1
+        ON e2.base_entity_id = c1.baseentityid
+    WHERE t1.group_identifier = jurisdictions.jurisdiction_id
+    AND t1.task_for = locations.id
+    AND t1.plan_identifier = plans.identifier
+    AND t1.code IN ('Structure_Visited', 'Structure Visited')
+    AND t1.status IN ('Completed')
+) AS client_info_query ON true
+-- lets investigate if we can enable this section
```

Comment:
This part if commented out, because I believe it can be made to work if we ensure that Canopy syncs 100% correctly.